pkg/asset/installconfig/azure/validation.go

```diff
@@ -995,3 +999,46 @@ func checkBootDiagnosticsURI(client API, diag *aztypes.BootDiagnostics, region s
 	}
 	return nil
 }
+
+func validateCustomSubnetsAndNatGateways(client API, fldPath *field.Path, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+	subnetSpec := ic.Azure.Subnets
+	virtualNetwork := ic.Azure.VirtualNetwork
+	networkResourceGroupName := ic.Azure.NetworkResourceGroupName
+
+	subnetList := map[string]*aznetwork.Subnet{}
```

Comment:
Nit: Can we rename this `vnetSubnetList`? That way it is easy to distinguish between the subnets within the virtual network and the ones provided as the new configuration.